src/lib/rpc/client.ts

```diff
@@ -106,3 +109,20 @@ export const findFirstUserAccess = async (token: string) => {
 
   return userInfo.access[0];
 };
+
+/**
+ * Get the tenant and access token for authenticated requests.
+ * This function will redirect to auth if the token is not available.
+ */
+export async function getTenantAndToken() {
```

Comment:
Low level libraries should not make redirect decisions. If this is required, move this to `src/lib/session` or somewhere which is more appropriate and pass args such as `onFailure` so that caller can decide where to redirect on failure